src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -734,14 +784,108 @@ subroutine read_fv3_netcdf_guess(fv3filenamegin)
+        call gsi_fv3ncdf_read(tracers,'GRAUPEL','graupel',ges_qg,mype_qg)
+        call gsi_fv3ncdf_read(tracers,'RAIN_NC','rain_nc',ges_qnr,mype_qnr)
+      else
+         write(6,*) "FV3 IO READ for 'fv3sar_bg_opt == 0 are only available for now in direct reflectivity DA"
+         stop
+      end if
+
 !!  tsen2tv  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-    do k=1,nsig
-       do j=1,lon2
-          do i=1,lat2
-             ges_tv(i,j,k)=ges_tsen(i,j,k,it)*(one+fv*ges_q(i,j,k))
+      do k=1,nsig
+        do i=1,lon2
+          do j=1,lat2
+             ges_tv(j,i,k)=ges_tsen(j,i,k,it)*(one+fv*ges_q(j,i,k))
+!            Apply power transform if option is ON 
+             if (l_use_cvpqx) then
+                if ( l_use_cvpqx_pval .eq. 0. ) then ! CVlogq
+                   if (ges_tsen(j,i,k,it) .gt. 274.15) then
+                      qr_min=2.9E-6_r_kind
+                      qr_thrshd=qr_min * one_tenth
+                      qs_min=0.1E-9_r_kind
+                      qs_thrshd=qs_min
+                      qg_min=3.1E-7_r_kind
+                      qg_thrshd=qg_min * one_tenth
+                   else if (ges_tsen(j,i,k,it) .le. 274.15 .and. ges_tsen(j,i,k,it) .ge. 272.15) then
+                      qr_min=2.0E-6_r_kind
+                      qr_thrshd=qr_min * one_tenth
+                      qs_min=1.3E-7_r_kind
+                      qs_thrshd=qs_min * one_tenth
+                      qg_min=3.1E-7_r_kind
+                      qg_thrshd=qg_min * one_tenth
+                   else if (ges_tsen(j,i,k,it) .lt. 272.15) then
+                      qr_min=0.1E-9_r_kind
+                      qr_thrshd=qr_min
+                      qs_min=6.3E-6_r_kind
+                      qs_thrshd=qs_min * one_tenth
+                      qg_min=3.1E-7_r_kind
+                      qg_thrshd=qg_min * one_tenth
+                   end if
+
+                   if ( ges_qr(j,i,k) .le. qr_thrshd )  ges_qr(j,i,k) = qr_min
+                   if ( ges_qs(j,i,k) .le. qs_thrshd )  ges_qs(j,i,k) = qs_min
+                   if ( ges_qg(j,i,k) .le. qg_thrshd )  ges_qg(j,i,k) = qg_min
+
+!                  convert hydrometer variables (qr,qs and qg) to log
+                   if(mype==0 .AND. i==3 .AND. j==3 .AND. k==3)then
+                       write(6,*)'read_fv3_netcdf_guess: ',     &
+                           ' reset zero of qr/qs/qg to specified values (~0dbz)', &
+                           'before log transformation. (for dbz assimilation)'
+                       write(6,*)'read_fv3_netcdf_guess: convert qr/qs/qg to log.'
+                   end if
+                   ges_qr(j,i,k) = log(ges_qr(j,i,k))
+                   ges_qs(j,i,k) = log(ges_qs(j,i,k))
+                   ges_qg(j,i,k) = log(ges_qg(j,i,k))
+
+                else if ( l_use_cvpqx_pval .gt. 0. ) then   ! CVpq
+!                  convert cloud hydrometer variables (qr,qs and qg) to log
+                   if(mype==0 .AND. i==3 .AND. j==3 .AND. k==3)then
```

Comment:
convert this kind of write to comments or move them out of the loop.

Reply:
I modified the code according to your comments. I took out these codes from loops and make them print one time at the start of the routine using if(mype==0).

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-4c910ebd862ea8bea4024bbfa73835e643efb4091155d4e8ef121388b14ab6ffR2797-R2811